plugin/execute_command.py

```diff
@@ -0,0 +1,56 @@
+import sublime
+from .core.registry import client_for_view, LspTextCommand
+from .core.settings import client_configs
+from .core.protocol import Request
+from .core.logging import debug
+from .core.rpc import Client
+
+try:
+    from typing import List, Optional, Dict, Any
+    assert List and Optional and Dict and Any
+except ImportError:
+    pass
+
+
+class LspExecuteCommand(LspTextCommand):
+    def __init__(self, view):
+        super().__init__(view)
+
+    def run(self, command_name) -> None:
+        client = client_for_view(self.view)
+        if not client:
+            self.view.show_popup("No connection with server")
+        else:
+            self._command_names = []   # type: List[str]
+            self._command_args = dict()   # type: Dict[str, Dict[str, Any]]
+            for config in client_configs.all:
+                for command in config.commands:
+                    self._command_names.append(command.name)
+                    self._command_args[command.name] = command.args
+
+            # TODO handle when command name is passed
```

Comment:
Can you clarify what this means?

Reply:
to my understanding the run method can take extra arguments, I want allowing the user to create a key binding to this sublime command directly (without having the open the quick pallet) and pass directly the LSP command name and maybe the args like 
```
  {"keys": ["f10"], "command": "lsp_execute", "args":{"command_name": "build-import"}}
```
So far this leads to 
```
Traceback (most recent call last):
  File "/opt/sublime_text/sublime_plugin.py", line 1067, in run_
    return self.run(edit, **args)
TypeError: run() got multiple values for argument 'command_name'
```

Reply:
Okay I see. The signature of `run` should I think be

```
def run(self, edit, command_name: str) -> None:
   ...
```
Because it is a sublime_plugin.TextCommand. These types of commands receive an `edit` object with which one can edit the view (we don't use it here though).

Reply:
tried it already and it get `plugin/execute_command.py:19: error: Signature of "run" incompatible with supertype "TextCommand"` with mypy

using `def run(self, edit, **command_name) -> None:` works but now is a Dict[str, Any] which defeats the purpose :(

Reply:
extra arguments need default value so this did the trick
```
    def run(self, edit, command_name: 'Optional[str]'=None, command_args: 'Dict[str, Any]'=dict()) -> None:
```